x/sync/manager.go

```diff
@@ -234,38 +274,47 @@ func (m *Manager) close() {
 	})
 }
 
-// Processes [item] by fetching and applying a change or range proof.
-// Assumes [m.workLock] is not held.
-func (m *Manager) doWork(ctx context.Context, work *workItem) {
-	defer func() {
-		m.workLock.Lock()
-		defer m.workLock.Unlock()
+func (m *Manager) finishWorkItem() {
+	m.workLock.Lock()
+	defer m.workLock.Unlock()
 
-		m.processingWorkItems--
-		m.unprocessedWorkCond.Signal()
-	}()
+	m.processingWorkItems--
+	m.unprocessedWorkCond.Signal()
+}
+
+// Processes [item] by fetching a change or range proof.
+func (m *Manager) doWork(ctx context.Context, work *workItem) {
+	// Backoff for failed requests accounting for time this job has already
+	// spent waiting in the unprocessed queue
+	backoff := calculateBackoff(work.attempt)
```

Comment:
Not optimal since this worker goroutine is going to be waiting until this timeout passes and ideally we would prioritize requests that are always executable, but felt like something more complex would make the diff hard to review since the scope of this PR is just trying to get rid of the blocking networking calls